src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs

```diff
@@ -168,7 +168,7 @@ private async Task<(Document document, bool addedFields)> GenerateThisOrBaseDele
                 var isThis = namedType.Equals(_state.TypeToGenerateIn);
                 var delegatingArguments = syntaxFactory.CreateArguments(delegatedConstructor.Parameters);
                 var baseConstructorArguments = isThis ? default : delegatingArguments;
-                var thisConstructorArguments = isThis ? delegatingArguments : default;
+                var thisConstructorArguments = isThis && !_state.IsConstructorInitializerGeneration ? delegatingArguments : default;
```

Comment:
Note: we still want to delegate to to another constructor in the same type.  You just don't want to delegate to the constructor that you're being invoked from.  

We also want tests for this.  For example, if i have this:

```c#
class C
{
    public C(int x) { }

    public C(int x, int y, int x) : this(x, y) { }
}
```

This should generate:

```c#
class C
{
    public C(int x) { }

    public C(int x, int y) : this(x) { }

    public C(int x, int y, int x) : this(x, y) { }
}
```